src/main/kotlin/org/rust/lang/core/completion/RsAttributeCompletionProvider.kt

```diff
@@ -51,8 +52,9 @@ object RsAttributeCompletionProvider : RsCompletionProvider() {
         onExternCrate to "macro_use no_link",
         onMod to "no_implicit_prelude path macro_use",
         onFn to "main start test cold naked export_name link_section lang inline track_caller " +
-            "panic_handler must_use",
+            "panic_handler must_use target_feature()",
         onTestFn to "should_panic ignore",
+        onProcMacroFn to "proc_macro proc_macro_derive() proc_macro_attribute",
```

Comment:
Let's add tests to ensure that we provide completion for these attributes only in proper places
See `RsAttributeCompletionTest`

Reply:
How do I run a completion test for a `proc-macro` crate? I couldn't find how I can create the relevant Cargo project.

Reply:
@afetisov 
```kotlin
@ProjectDescriptor(WithDependencyRustProjectDescriptor::class)
fun `test resolve use item inside proc macro crate`() = stubOnlyResolve("""
//- dep-proc-macro/lib.rs
    #[proc_macro]
    pub fn example_proc_macro(item: TokenStream) -> TokenStream { item }
//- lib.rs
    use dep_proc_macro::example_proc_macro;
""")
```

Reply:
@vlad20012 This won't work. I'm basing the presence of proc macro attributes on the type of the crate, and it's defined in Cargo.toml. How do I make a test proc macro crate?

Reply:
Since it is `dep-proc-macro/lib.rs` and *not* `dep-lib` (as usual), it's crate type is proc macro, so it will work

Reply:
You can find all hardcoded packages/targets/paths in the `WithDependencyRustProjectDescriptor` class source

Reply:
So it's based on the dependency name? Cool... but also super cryptic. It would be really awesome if someone added proper documentation to the testing framework. It's not documented both at the plugin and the platform level, and quite hard to make sense of.

Reply:
> So it's based on the dependency name

Well, yes and no. `WithDependencyRustProjectDescriptor` just offers a hardcoded project structure. A test that uses `WithDependencyRustProjectDescriptor` always have `dep-proc-macro` dependency, as well as other 15 dependencies, regardless of what you write in the test itself